core/block.js

```diff
@@ -73,7 +73,7 @@ Blockly.Block = function(workspace, prototypeName, opt_id) {
    * @private
    */
   this.disabled = false;
-  /** @type {string|!Function} */
+  /** @type {!string|!function(): (!string|!Function)|!{tooltip}} */
```

Comment:
If you're asking about the type as a whole, I just figured this out from staring at the existing code for a long time hehe. It can be a string, it can be a function which returns (a) a string or (b) arbitrarily nested other functions, or it can be an object which has a .tooltip property, which has one of the other types.

If you're just asking about the !{tooltip} part, I think that's how you're supposed to say that it could be an object with .tooltip property.

The !{tooltip} used to be an SVGElement, but it wasn't actually required to be an svg element, and that requirement wasn't maintained, so I changed it to an object containing a tooltip property.